src/solvers/file_solution_history.C

```diff
@@ -53,10 +53,21 @@ void FileSolutionHistory::find_stored_entry(Real time)
   // For the key right before the lower bound
   stored_solutions_iterator lower_bound_it_decremented;
 
-  // If we are at end, we are creating a new entry, nothing more to do
+  // If we are at end, we could be creating a new entry (depends on the storing bool), return
+  // Otherwise, get a decremented iterator for the sandwich test
   if(lower_bound_it == stored_solutions.end())
   {
-    return;
+    if(storing)
+    {
+      return;
```

Comment:
So at this point we haven't updated stored_sols?  Isn't that potentially a bug if we don't have an empty history?